src/transformers/modeling_auto.py

```diff
@@ -210,6 +212,46 @@
     ]
 )
 
+MODEL_FOR_CAUSAL_LM_MAPPING = OrderedDict(
+    [
+        (BertConfig, BertLMHeadModel),
+        (OpenAIGPTConfig, OpenAIGPTLMHeadModel),
+        (GPT2Config, GPT2LMHeadModel),
+        (TransfoXLConfig, TransfoXLLMHeadModel),
+        (XLNetConfig, XLNetLMHeadModel),
+        (
+            XLMConfig,
+            XLMWithLMHeadModel,
+        ),  # XLM can be MLM and CLM => model should be split similar to BERT; leave here for now
```

Comment:
Eh, I hadn't realized but `XLMWithLMHead` does not compute the loss when provided with labels because of this. Definitely needs a split and both splits to return the proper loss if provided with labels